### PR TITLE
Add basic examples

### PR DESCRIPTION
**Pull request by trentmwillis**
This takes the majority of the leaks in the original `my-c` example and splits them out into a more complicated system that better mirrors a real world application. The app still needs some styling and additional functionality, but this should serve as a base line.

### CODE REVIEW

File: app/components/list-view.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  didReceiveAttrs() {
```
Comment (stefanpenner) [inline app/components/list-view.js]: Why wouldn't this be done in `didInsertElement`, this hooks seems strange to care about DOM stuff?

File: app/components/my-c.js [hunk 2]
```diff
@@ -1,47 +0,0 @@
...
-    this._super(...arguments);
-    this.scrollCount = 0;
-    this.other();
-    SCROLL_COUNTER.set(this, { count: 0 });
-  },
-
-  other() {
-    this.foo = () => this;
-    window.foo = function() {
-      return SOME_STRING;
```
Comment (stefanpenner) [inline app/components/my-c.js]: is this type of leak still represented?

Comment (trentmwillis) [inline app/components/my-c.js]: I could not think of a non-contrived example, but still have it as a todo item

File: app/components/list-view.js (hunk 1 above)
Comment (trentmwillis) [inline app/components/list-view.js]: Doesn't really matter to me. Will change.

File: app/components/my-c.js (hunk 2 above)
Comment (trentmwillis) [inline app/components/my-c.js]: @stefanpenner added this to cover this type of leak: https://github.com/trentmwillis/memory-leak-examples/blob/3d47e12f46964ab78f0f20c5e66c9cd8ce843e9f/app/utils/component-debug.js

Seem good?

**Pull request merged**